src/github.com/couchbase/sync_gateway/base/bucket.go

```diff
@@ -55,15 +55,56 @@ type BucketSpec struct {
 
 // These are used by CBGT to determine the sharding factor and other properties
 type FeedParams struct {
-	MaxPartitionsPerPIndex int `json:"max_partitions_per_pindex"`
-	NumReplicas            int `json:"num_replicas"`
+	NumShards   int `json:"num_shards"`
+	NumReplicas int `json:"num_replicas"`
```

Comment:
What's your take on best practices around int vs. int8/16/32/64 vs. uint8/16/32/64.

Since int goes to a different size based on the compiler, I thought the sized ints were preferable - but that can end up with a lot more size casting in the code.

Also - because the use of uint32/64 is so common in go-couchbase  (mainly because it needs to be formatted for the memcached binary protocol), I've gotten into the habit of using a sized uint everywhere. However, I've also seen people feeling strongly that uint shouldn't be used only to specify non-negative functional behaviour, as it can lead to unexpected errors during arithmetic. 

I'd be interested to hear your take.